src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs

```diff
@@ -0,0 +1,134 @@
+        public override UpdateRowSource UpdatedRowSource { get => _inner.UpdatedRowSource; set => _inner.UpdatedRowSource = value; }
+        protected override DbConnection DbConnection { get => _inner.Connection; set => _inner.Connection = value; }
+        protected override DbTransaction DbTransaction { get => _inner.Transaction; set => _inner.Transaction = value; }
+        protected override DbParameterCollection DbParameterCollection => _inner.Parameters;
+
+        public override void Cancel() => _inner.Cancel();
+        public override void Prepare() => _inner.Prepare();
+        protected override DbParameter CreateDbParameter() => _inner.CreateParameter();
+
+        private void SpanDefaults(OpenTracing.ISpan span)
+        {
+            var operation = CommandText.Split().First().ToUpper();
+            span.SetOperationName(operation);
+
+            span.SetTag("event.id", Guid.NewGuid().ToString());
+            span.SetTag("event.origin", "system-data");
+            span.SetTag("resource.type", "database");
+            span.SetTag("resource.name", DbConnection.Database);
+            span.SetTag("resource.operation", operation);
+            span.SetTag("sql.driver", DbConnection.GetType().FullName);
+            span.SetTag("sql.statement", CommandText);
+            span.SetTag("sql.connection_string", DbConnection.ConnectionString);
+        }
+
+        public override int ExecuteNonQuery()
+        {
+            using (var scope = GlobalTracer.Instance
+                            .BuildSpan("")
+                            .WithStartTimestamp(DateTime.Now)
+                            .StartActive(finishSpanOnDispose: true))
+            {
+                int affected_rows = -1;
+                var span = scope.Span;
+
+                SpanDefaults(span);
+                span.SetTag("sql.table_name", ""); // check other instrum to see how this is obtained
+                span.SetTag("sql.parameters", ""); // check other instrum to see how this is obtained
+
+                try { affected_rows = _inner.ExecuteNonQuery(); }
+                catch (Exception e)
+                {
+                    span.AddException(e);
+                    throw;
+                }
+
+                span.SetTag("sql.cursor_row_count", affected_rows);
+                return affected_rows;
+            }
+        }
+
+        public override object ExecuteScalar()
+        {
+            using (var scope = GlobalTracer.Instance
+                            .BuildSpan("")
+                            .WithStartTimestamp(DateTime.Now)
+                            .StartActive(finishSpanOnDispose: true))
+            {
+                var span = scope.Span;
+                object result = null;
+
+                SpanDefaults(span);
+                span.SetTag("sql.table_name", ""); // check other instrum to see how this is obtained
```

Comment:
Can we will this info from the query itself / connection object?

Reply:
yeah didnt get to it yet, we need to kindof parse the sql to get it